src/SnD.ApiClient/Crystal/Base/ICrystalClient.cs

```diff
@@ -25,4 +25,16 @@ public interface ICrystalClient
     /// <param name="cancellationToken">Cancellation token</param>
     /// <returns>RunResult instance</returns>
     public Task<RunResult> GetResultAsync(string algorithm, string requestId, CancellationToken cancellationToken = default);
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Awaits a run until it completes with any result or runs out of time set via cancellationToken.
+    /// For some cases this can return a failed result when the algorithm has not been actually executed, for example,
+    /// when a submission was lost.
+    /// </summary>
+    /// <param name="algorithm">Algorithm name</param>
+    /// <param name="requestId">Request ID received form <see cref="CreateRunAsync"/></param>
+    /// <param name="cancellationToken">Cancellation token for the operation timeout.</param>
+    /// <param name="pollIntervalSeconds">Poll interval to check for run results in seconds.</param>
+    /// <returns>RunResult instance</returns>
+    public Task<RunResult> AwaitRunAsync(string algorithm, string requestId, CancellationToken cancellationToken, int pollIntervalSeconds = 1);
```

Comment:
```suggestion
    public Task<RunResult> AwaitRunAsync(string algorithm, string requestId, int pollIntervalSeconds = 1, CancellationToken cancellationToken=default);
```
Also, change type of `pollIntervalSeconds` to `TimeSpan`.